tuna/rocmlir/import_configs.py

```diff
@@ -28,141 +28,27 @@
-        datatype = ""
-
-      # Skip direction if already in
-      direction = f"{direction} "
-      if "-F" in line:
-        direction = ""
-
-      # Skip filter layout if already in
-      filter_layout = f"-f {layout} "
-      if "-f" in line:
-        filter_layout = ""
-
-      # Skip input layout if already in
-      input_layout = f"-I {layout} "
-      if "-I" in line:
-        input_layout = ""
-
-      # Skip output layout if already in
-      output_layout = f"-O {layout} "
-      if "-O" in line:
-        output_layout = ""
-
-      one_config = f"{datatype}{direction}{filter_layout}{input_layout}{output_layout}{line}"
-      if one_config not in configs:
-        configs.append(one_config)
-
-  return configs
-
-
-def parse_line(line, dbt):
-  """Parse a command-line-style conv config into a ConvolutionConfig object."""
-
-  print(f"Parsing line {line}")
-
-  # '-t 1' means 'enable timing' which is confused with -t for type.
-  line = line.replace("-t 1", "")
-
-  # Convert the line ("-n 256 -c 1024 -H 14 ...") to dict of flag and value.
-  i = iter(line.split())
-  options = dict(zip(i, i))
-  #  print(f"options = {options}")
-
-  # Mapping of flag to field name.
-  # -F 1 -n 2 -c 1280 -H 32 -W 32 -k 640 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
-  fields = {
-      '-F': 'direction',
-      '-f': 'fil_layout',
-      '-I': 'in_layout',
-      '-O': 'out_layout',
-      '-n': 'batchsize',
-      '-c': 'in_channels',
-      '-H': 'in_h',
-      '-W': 'in_w',
-      '-k': 'out_channels',
-      '-y': 'fil_h',
-      '-x': 'fil_w',
-      '-p': 'pad_h',
-      '-q': 'pad_w',
-      '-u': 'conv_stride_h',
-      '-v': 'conv_stride_w',
-      '-l': 'dilation_h',
-      '-j': 'dilation_w',
-      '-g': 'group_size',
-      '-m': None,
-      '-t': 'data_type'
-  }
-  # kernel-repeats has no flag, but perfRunner.py uses 5.
-  # ConvConfiguration.fromCommandLine accepts but skips -m and -t.
-  #   -m is operation and -t is (I think) -time from MIOpenDriver.
-  # data_type is inferred from operation -- conv is f32, convfp16 is f16,
-  #   convbfp16 is bf16, convint8 is i8
-
-  config = dbt.config_table(kernel_repeats=1)
-  for flag, value in options.items():
-    field = fields[flag]
-    if field:
-      setattr(config, field, value)
-
-  return config
+from tuna.rocmlir.rocmlir_tables import RocMLIRDBTables, RocMLIRDBTablesConv, RocMLIRDBTablesGEMM
 
 
 def import_cfgs(args: argparse.Namespace, dbt: RocMLIRDBTables,
                 logger: logging.Logger):
   """import configs to mysql from file with driver invocations"""
   connect_db()
-
-  configs = get_conv_configurations(os.path.expanduser(args.file_name))
+  configs = dbt.get_configurations(os.path.expanduser(args.file_name))
+  print(configs, file=sys.stderr)
   for line in configs:
     try:
-      config = parse_line(line, dbt)
-
+      config = dbt.config_table()
+      config.parse_line(line)
       with DbSession() as session:
         try:
```

Comment:
Yes, every time the 'with' statement is entered a new session is being created. So moving this outside the loop would reduce the overhead.

Reply:
Done.  Additional slight change to make a new config-table object on each iteration, because session.add won't add the same object twice, even if its attributes have changed and even if it has committed it.